### ISSUE TITLE
Feature or Bug: Is OGA support the feature all '''DeviceSpan<int32_t>& next_tokens''' are '''model.pad_token_id'''?

### ISSUE THREAD

**Issue by wangming-amd** [microsoft/onnxruntime-genai#1297]
**Describe the bug**

Is OGA support the feature all '''DeviceSpan<int32_t>& next_tokens''' are '''model.pad_token_id'''?

If the answer is Yes, there maybe bug in the code. if the answer is No, do you have plan to support this feature? or how do you deal with this scene?

If all '''DeviceSpan<int32_t>& next_tokens''' are '''model.pad_token_id''':
in the following code, after the end of while loop,  token_index will be -1, and input_sequence_lengths[b] will be 0;
https://github.com/microsoft/onnxruntime-genai/blob/b247a9370565d9596d442c9f22a552582aecff98/src/models/logits.cpp#L96-L100

then go to the following code, token_index = -1; then in L55, the first parameter of logits_raw.subspan(),
'''(vocab_index * seq_length + token_index * vocab_size) * element_size''' maybe = (0 * seq_length + **(-1 )** * vocab_size) * element_size = **-1** * vocab_size * element_size, it's a negative value.
https://github.com/microsoft/onnxruntime-genai/blob/b247a9370565d9596d442c9f22a552582aecff98/src/models/logits.cpp#L52-L58

in the subspan(), the parameter datatype is size_t,  it's unsigned, so if the first parameter is a negative value, it maybe very large as to size_t, then L416 will report error and lead to crash.
using size_type        = size_t;

<img width="836" alt="Image" src="https://github.com/user-attachments/assets/2da2d2f1-4c71-4bff-b734-76a39d1c342c" />

<img width="207" alt="Image" src="https://github.com/user-attachments/assets/0d12fd0a-8223-49c2-be54-0c6a4df07756" />

**Comment by RyanUnderhill** [comment]
@aciddelgado You probably know the most about this new continuous decoding logic, we should support this edge case. I can add a unit test but what do you think the best behavior is when it happens?

**Comment by buyuliushen** [comment]
Hi @RyanUnderhill  @aciddelgado ， any update about this?
